libs/librrgraph/src/io/rr_graph_uxsdcxx_serializer.h

```diff
@@ -460,19 +460,26 @@ class RrGraphSerializer final : public uxsd::RrGraphBase<RrGraphContextTypes> {
         //
         // If the switch name is not present in the architecture, generate an
         // error.
+        // If the graph is written when flat-routing is enabled, the types of the switches inside of the rr_graph are also
+        // added to the XML file. These types are not added to the data structure that contain arch switch types. They are added to all_sw_inf under dvice context.
```

Comment:
Because of collapsing node chains, some switch types are determined when the collapsing analysis is being done, and it is not possible to get all switch types required by flat-routing only by analyzing the architecture file. Unless you want to do the collapsing analysts in all cases.